src/careamics/prediction_utils/prediction_outputs.py

```diff
@@ -0,0 +1,165 @@
+"""Module containing functions to convert prediction outputs to desired form."""
+
+from typing import Any, List, Literal, Tuple, Union, overload
+
+import numpy as np
+from numpy.typing import NDArray
+
+from ..config.tile_information import TileInformation
+from .stitch_prediction import stitch_prediction
+
+
+def convert_outputs(
+    predictions: List[Any], tiled: bool
+) -> Union[List[NDArray], NDArray]:
+    """
+    Convert the outputs to the desired form.
+
+    Parameters
+    ----------
+    predictions : list
+        Predictions that are output from `Trainer.predict`.
+    tiled : bool
+        Whether the predictions are tiled.
+
+    Returns
+    -------
+    list of numpy.ndarray or numpy.ndarray
+        List of arrays with the axes SC(Z)YX. If there is only 1 output it will not
+        be in a list.
+    """
+    if len(predictions) == 0:
+        return predictions
+
+    # this layout is to stop mypy complaining
+    if tiled:
+        predictions_comb = combine_batches(predictions, tiled)
+        # remove sample dimension (always 1) `stitch_predict` func expects no S dim
+        tiles = [pred[0] for pred in predictions_comb[0]]
+        tile_infos = predictions_comb[1]
+        predictions_output = stitch_prediction(tiles, tile_infos)
+    else:
+        predictions_output = combine_batches(predictions, tiled)
+
+    # TODO: add this in? Returns output with same axes as input
```

Comment:
I think there are some inconsistencies with output dims depending on whether the prediction was tiled or not. If the prediction was tiled then because the function `stitch_prediction` doesn't expect the S dim, then if there was only one image, then the output will also not have an S dim. However, if the prediction is not tiled, then the S dimension is not removed, even if there is only one image.

I have left it like this because I am pretty sure it is recreating how the code was previously behaving, but I can merge this PR and then create a new PR to fix this issue.

Reply:
Let's make an issue out of it so that we don't loose track of it! Also with respect to the list or not list return value.

I think for now it is ok to merge!